### ISSUE TITLE
"Just-in-time loading and compiling the CUDA kernels of SRU was unsuccessul"

### ISSUE THREAD

**Issue by zhihanyang2022** [asappresearch/sru#190]
Does anyone know how to fix this error?

```
>>> from sru import SRU
/home/hnguyen/miniconda3/lib/python3.8/site-packages/sru/cuda_functional.py:23: UserWarning: Just-in-time loading and compiling the CUDA kernels of SRU was unsuccessful. Got the following error:
Error building extension 'sru_cuda': [1/2] /usr/local/cuda-10.1/bin/nvcc --generate-dependencies-with-compile --dependency-output sru_cuda_kernel.cuda.o.d -DTORCH_EXTENSION_NAME=sru_cuda -DTORCH_API_INCLUDE_EXTENSION_H -DPYBIND11_COMPILER_TYPE=\"_gcc\" -DPYBIND11_STDLIB=\"_libstdcpp\" -DPYBIND11_BUILD_ABI=\"_cxxabi1011\" -isystem /home/hnguyen/miniconda3/lib/python3.8/site-packages/torch/include -isystem /home/hnguyen/miniconda3/lib/python3.8/site-packages/torch/include/torch/csrc/api/include -isystem /home/hnguyen/miniconda3/lib/python3.8/site-packages/torch/include/TH -isystem /home/hnguyen/miniconda3/lib/python3.8/site-packages/torch/include/THC -isystem /usr/local/cuda-10.1/include -isystem /home/hnguyen/miniconda3/include/python3.8 -D_GLIBCXX_USE_CXX11_ABI=0 -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_BFLOAT16_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --expt-relaxed-constexpr -gencode=arch=compute_70,code=compute_70 -gencode=arch=compute_70,code=sm_70 --compiler-options '-fPIC' -std=c++14 -c /home/hnguyen/miniconda3/lib/python3.8/site-packages/sru/csrc/sru_cuda_kernel.cu -o sru_cuda_kernel.cuda.o 
FAILED: sru_cuda_kernel.cuda.o 
/usr/local/cuda-10.1/bin/nvcc --generate-dependencies-with-compile --dependency-output sru_cuda_kernel.cuda.o.d -DTORCH_EXTENSION_NAME=sru_cuda -DTORCH_API_INCLUDE_EXTENSION_H -DPYBIND11_COMPILER_TYPE=\"_gcc\" -DPYBIND11_STDLIB=\"_libstdcpp\" -DPYBIND11_BUILD_ABI=\"_cxxabi1011\" -isystem /home/hnguyen/miniconda3/lib/python3.8/site-packages/torch/include -isystem /home/hnguyen/miniconda3/lib/python3.8/site-packages/torch/include/torch/csrc/api/include -isystem /home/hnguyen/miniconda3/lib/python3.8/site-packages/torch/include/TH -isystem /home/hnguyen/miniconda3/lib/python3.8/site-packages/torch/include/THC -isystem /usr/local/cuda-10.1/include -isystem /home/hnguyen/miniconda3/include/python3.8 -D_GLIBCXX_USE_CXX11_ABI=0 -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_BFLOAT16_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --expt-relaxed-constexpr -gencode=arch=compute_70,code=compute_70 -gencode=arch=compute_70,code=sm_70 --compiler-options '-fPIC' -std=c++14 -c /home/hnguyen/miniconda3/lib/python3.8/site-packages/sru/csrc/sru_cuda_kernel.cu -o sru_cuda_kernel.cuda.o 
nvcc fatal   : Unknown option '-generate-dependencies-with-compile'
ninja: build stopped: subcommand failed.

warnings.warn("Just-in-time loading and compiling the CUDA kernels of SRU was unsuccessful. "
```

**Comment by zhihanyang2022** [comment]
Digging into the documentation for nvcc eventually resolved my issue. If you look into the [doc](https://docs.nvidia.com/cuda/archive/10.1/cuda-compiler-driver-nvcc/index.html) for NVCC associated with CUDA 10.1, then you indeed can't find the option `-generate-dependencies-with-compile` (just do a ctrl-F webpage text search). But if you look into the [doc](https://docs.nvidia.com/cuda/archive/10.2/cuda-compiler-driver-nvcc/index.html) for NVCC associated with CUDA 10.2, then you will find this option. 

In short, changing CUDA version from 10.1 to 10.2 resolved this error.

Cheers.

**Comment by v-nhandt21** [comment]
> 

I am using CUDA=11.1 in conda, how can I handle with this problem
